### PR TITLE
Disable x87 floating point for 32-bit code generation

### PR DESCRIPTION
**Pull request by 0xdaryl**
The minimum target processor level for OMR code generation is Pentium 4,
which has support for SSE2.  All modern operating systems are assumed
to support preserving the SSE state.  Disable x87 code generation in
preparation for removal.

Issue: #946

Signed-off-by: Daryl Maier <maier@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins build all

**Comment by 0xdaryl** [comment]
WIPing this until downstream testing completes.

**Comment by 0xdaryl** [comment]
Jenkins build all

**Comment by 0xdaryl** [comment]
Un-WIPing as downstream testing looks fine.

File: compiler/x/codegen/OMRCodeGenerator.cpp
Comment (BradleyWood) [inline compiler/x/codegen/OMRCodeGenerator.cpp:235]: The SSE support check seems to be covered by the assertion.

Comment (0xdaryl) [reply compiler/x/codegen/OMRCodeGenerator.cpp:235]: Yes.  This PR is about doing the minimal work to disable x87 by default in OMR to enable further refactoring.  The goal of the follow-on work to deprecate and remove the x87 code should be to smooth all of this out.

**Review by fjeremic — APPROVED** [review]

File: compiler/x/codegen/OMRCodeGenerator.cpp
Comment (fjeremic) [reply compiler/x/codegen/OMRCodeGenerator.cpp:235]: @BradleyWood will wait for your approval if you're satisfied with Daryl's reasoning for this review. Thanks!

Comment (BradleyWood) [reply compiler/x/codegen/OMRCodeGenerator.cpp:235]: Sure, go ahead. We will address in future work.

**Pull request merged**
